src/controllers/endorseCompanyClosureCertificate.controller.ts

```diff
@@ -1,35 +1,62 @@
+import { BAD_REQUEST, OK } from 'http-status-codes'
 import { inject } from 'inversify'
-import { controller, httpGet, httpPost } from 'inversify-express-utils'
+import { controller, httpGet, httpPost, requestBody } from 'inversify-express-utils'
+import { RedirectResult } from 'inversify-express-utils/dts/results'
 import BaseController from './base.controller'
 
 import DissolutionApprovalModel from 'app/models/form/dissolutionApproval.model'
-import { ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI, REDIRECT_GATE_URI, } from 'app/paths'
+import EndorseCertificateFormModel from 'app/models/form/endorseCertificateFormModel'
+import Optional from 'app/models/optional'
+import DissolutionSession from 'app/models/session/dissolutionSession.model'
+import ValidationErrors from 'app/models/view/validationErrors.model'
+import { ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI, PAYMENT_URI, } from 'app/paths'
+import formSchema from 'app/schemas/endorseCertificate.schema'
+import DissolutionService from 'app/services/dissolution/dissolution.service'
 import SessionService from 'app/services/session/session.service'
 import TYPES from 'app/types'
+import FormValidator from 'app/utils/formValidator.util'
 
 interface ViewModel {
   approvalModel: DissolutionApprovalModel
+  errors?: ValidationErrors
 }
 
 @controller(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI, TYPES.SessionMiddleware, TYPES.AuthMiddleware, TYPES.CompanyAuthMiddleware)
 export class EndorseCompanyClosureCertificateController extends BaseController {
 
-  public constructor(@inject(SessionService) private session: SessionService) {
+  public constructor(@inject(SessionService) private session: SessionService,
+                    @inject(FormValidator) private validator: FormValidator,
+                    @inject(DissolutionService) private dissolutionService: DissolutionService) {
     super()
   }
 
   @httpGet('')
   public async get(): Promise<string> {
-    const approvalModel: DissolutionApprovalModel = this.session.getDissolutionSession(this.httpContext.request)!.approval!
+    return this.renderView()
+  }
 
-    const viewModel: ViewModel = {
-      approvalModel
+  @httpPost('')
+  public async post(@requestBody() body: EndorseCertificateFormModel): Promise<string | RedirectResult> {
+    const errors: Optional<ValidationErrors> = this.validator.validate(body, formSchema)
+    if (errors) {
+      return this.renderView(errors)
     }
-    return super.render('endorse-company-closure-certificate', viewModel)
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const userEmail: string = this.session.getUserEmail(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    await this.dissolutionService.approveDissolution(token, dissolutionSession, userEmail)
+
+    return this.redirect(PAYMENT_URI) // TODO - Change to Redirect once logic is there 🐓
```

Comment:
This should go to the Redirect Controller, and the redirect controller should navigate to the `PAYMENT_URI` temporarily.